src/app/modules/time-clock/components/entry-fields/entry-fields.component.ts

```diff
@@ -93,6 +107,20 @@ export class EntryFieldsComponent implements OnInit {
     this.store.dispatch(new entryActions.UpdateEntryRunning({ ...this.newData, ...this.entryForm.value }));
   }
 
+  onUpdateStartHour() {
+    const startDate = formatDate(this.activeEntry.start_date, 'yyyy-MM-dd', 'en');
+    const newHourEntered = new Date(`${startDate}T${this.entryForm.value.start_hour.trim()}`).toISOString();
+    const isEntryDateInTheFuture = moment(newHourEntered).isAfter(moment());
+    if (isEntryDateInTheFuture) {
+      this.toastrService.error('You cannot start a time-entry in the future');
+      this.entryForm.patchValue({ start_hour: this.newData.start_hour });
+      return;
+    }
+    this.entryForm.patchValue({ start_date: newHourEntered });
+    this.store.dispatch(new entryActions.UpdateEntryRunning({ ...this.newData, ...this.entryForm.value }));
+    this.store.dispatch(new entryActions.LoadActiveEntry());
```

Comment:
It seems we also need to update Summary. Add the following line after line 121:
`this.store.dispatch(new entryActions.LoadEntriesSummary());`

Reply:
Update tests accordingly.